packages/core/src/domain/sessionManagement.ts

```diff
@@ -48,10 +50,26 @@ export function startSessionManagement<TrackingType extends string>(
       persistSession(session, sessionCookie)
 
       // If the session id has changed, notify that the session has been renewed
-      if (isTracked && currentSessionId !== session.id) {
-        currentSessionId = session.id
+      if (isTracked && initialSessionState.id !== session.id) {
+        initialSessionState = { ...session }
```

Comment:
Since the `initialSessionState` is reaffected, maybe it should be called something else? I think `currentSessionState` (or `currentSession`, since the `State` suffix is not used for `session` and `retrievedSession` variables) is more accurate.

Reply:
hum, IMO `current` is a bit confusing since the `retrieved` state could be qualified of `current`
maybe `stored` or `inMemory` could be more accurate
wdyt?

Reply:
submitted with `inMemorySession` and `cookieSession`